python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -771,19 +715,16 @@ def __init__(cls, name, bases, this_dict):
         # create reference in schema to the pluginclass
         this_schema._pluginclass = cls
 
-        # store generated proxy class
-        setattr(cls, '_proxyClass', proxyClass)
-
-        # register plugin class
-        if hasattr(cls, '_declared_property'):
-            # if we've not even declared this we don't want to use it!
-            if not cls._declared_property('hidden') or cls._declared_property('enable_plugin'):
-                allPlugins.add(cls, cls._category, getName(cls))
+        # if we've not even declared this we don't want to use it!
+        if not cls._declared_property('hidden') or cls._declared_property('enable_plugin'):
+            allPlugins.add(cls, cls._category, getName(cls))
 
-            # create a configuration unit for default values of object properties
-            if not cls._declared_property('hidden') or cls._declared_property('enable_config'):
-                this_schema.createDefaultConfig()
+        # create a configuration unit for default values of object properties
+        if not cls._declared_property('hidden') or cls._declared_property('enable_config'):
+            this_schema.createDefaultConfig()
 
+        # store generated proxy class
+        setattr(cls, '_proxyClass', GPIProxyClassFactory(name, cls))
```

Comment:
It can't be moved just yet as there's currently no other way to call `GPIProxyClassFactory()`. What I'm planning on adding (and have started work on) is a decorator for `GangaObject` subclasses which adds this and registers them in the GPI (similar to how Alex's system will work). For now I just wanted to combine all this to a single line.